### ISSUE TITLE
Cannot compile types with external ref through findValues

### ISSUE THREAD

**Issue by zifeo** [lloydmeta/enumeratum#96]
Version: `latest`

```scala
sealed abstract class A private (val value: Int) extends IntEnumEntry {
  val text: String
}

object A extends IntEnum[A] {

  val values = findValues

  case object A1 extends A(1) {
    val text = identity("something") // Error:(9, 16) object A1 has a value with the wrong type: something:class java.lang.String, instead of int.
    //val text = "something"
  }

  def identity(str: String) = str

}
```

No workaround known, blocker.

**Comment by lloydmeta** [comment]
I see, I think the problem is [here](https://github.com/lloydmeta/enumeratum/blob/master/macros/src/main/scala/enumeratum/ValueEnumMacros.scala#L175) where I crawl through the body looking for apply statements. There needs to be a check to make sure that this is a constructor application.

**Comment by lloydmeta** [comment]
Just released v1.5.5 with a fix for this issue. Thanks for reporting it :)

**Comment by zifeo** [comment]
Wow, thanks was fast! Thanks 👍
